gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -345,13 +345,18 @@ def self.maybe_run_sig_block_for_method(method)
     run_sig_block_for_key(key) if has_sig_block_for_key(key)
   end
 
-  def self.run_sig_block_for_method(method)
-    run_sig_block_for_key(method_to_key(method))
+  def self.run_sig_block_for_method(method, sig_block)
+    run_sig_block_for_key(method_to_key(method), sig_block)
   end
 
-  private_class_method def self.run_sig_block_for_key(key)
+  private_class_method def self.run_sig_block_for_key(key, sig_block = nil)
     blk = @sig_wrappers[key]
-    if !blk
+    if sig_block
+      if blk && blk != sig_block
+        raise "Bug: `sig` wrapper for #{key_to_method(key)} stored with wrong key"
+      end
+      blk = sig_block
```

Comment:
Is this line redundant? I think at this point `!(blk != sig_block)`?

Reply:
Should be more clear now hopefully! The sanity test I had complicated things too much.